src/deck/drivers/src/color_led_deck.c

```diff
@@ -46,23 +46,28 @@
 #include "led_deck_controller.h"
 
 
-static bool isInit = false;
-static uint8_t brightnessCorr = true;
-
-static uint32_t currentWrgb8888 = 0;
-static uint32_t wrgb8888 = 0;
-
-// Thermal status from deck
-static uint8_t deckTemperature = 0;
-static uint8_t throttlePercentage = 0;
-
-static void task(void* param);
-static paramVarId_t wrgbParamId;
-
-static DeckInfo* deck_info;
-
-static bool isInBootloader = false;
-static bool isInFirmware = true;
+// Context structure to hold per-instance state
+typedef struct {
+  bool isInit;
+  uint8_t brightnessCorr;
+  uint32_t currentWrgb8888;
+  uint32_t wrgb8888;
+  uint8_t deckTemperature;
+  uint8_t throttlePercentage;
+  uint8_t ledPosition;
+  uint16_t ledCurrent[4];  // LED current in milliamps for 4 channels (R, G, B, W)
+  paramVarId_t wrgbParamId;
+  uint8_t i2cAddress;
+  DeckInfo *deckInfo;
+  bool isInBootloader;
+  bool isInFirmware;
+} colorLedContext_t;
+
+// Two instances: bottom (index 0) and top (index 1)
```

Comment:
Create defines here for bottom and top which can be used later on in the file instead of 0/1 to specifically address each context.

Reply:
Fixed in 3a9f9b306e16eb86666fa67e7378c7c80e88598a